huaweicloud/services/evs/resource_huaweicloud_evs_volume_transfer.go

```diff
@@ -0,0 +1,189 @@
+)
+
+// @API EVS POST /v2/{project_id}/os-volume-transfer
+// @API EVS GET /v2/{project_id}/os-volume-transfer/{transfer_id}
+// @API EVS DELETE /v2/{project_id}/os-volume-transfer/{transfer_id}
+func ResourceVolumeTransfer() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVolumeTransferCreate,
+		ReadContext:   resourceVolumeTransferRead,
+		DeleteContext: resourceVolumeTransferDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"volume_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"auth_key": {
+				Type:      schema.TypeString,
+				Sensitive: true,
+				Computed:  true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceVolumeTransferCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg                         = meta.(*config.Config)
+		region                      = cfg.GetRegion(d)
+		createVolumeTransferHttpUrl = "v2/{project_id}/os-volume-transfer"
+		product                     = "evs"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+
+	createVolumeTransferPath := client.Endpoint + createVolumeTransferHttpUrl
+	createVolumeTransferPath = strings.ReplaceAll(createVolumeTransferPath, "{project_id}", client.ProjectID)
+	createVolumeTransferBodyParams := map[string]interface{}{
+		"transfer": map[string]interface{}{
+			"name":      d.Get("name"),
+			"volume_id": d.Get("volume_id"),
+		},
+	}
+	createVolumeTransferOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         createVolumeTransferBodyParams,
+	}
+
+	createResponse, err := client.Request("POST", createVolumeTransferPath, &createVolumeTransferOpt)
+	if err != nil {
+		return diag.Errorf("error creating EVS volume transfer: %s", err)
+	}
+
+	createRespBody, err := utils.FlattenResponse(createResponse)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	transferId := utils.PathSearch("transfer.id", createRespBody, "").(string)
+	if transferId == "" {
+		return diag.Errorf("error creating EVS volume transfer: ID is not found in API response")
+	}
+
+	d.SetId(transferId)
+	// The `auth_key` field is an important attribute and will only be returned when calling the create API.
+	// So it is necessary to make a no null judgment and set value here.
+	authKey := utils.PathSearch("transfer.auth_key", createRespBody, "").(string)
+	if authKey == "" {
+		return diag.Errorf("error creating EVS volume transfer: auth_key is not found in API response")
+	}
+	mErr := multierror.Append(nil, d.Set("auth_key", authKey))
+	if mErr.ErrorOrNil() != nil {
+		diag.FromErr(mErr)
+	}
```

Comment:
It is unnecessary using `multierror` here.
if err := d.Set("auth_key", authKey); err != nil {
  return diag.Errorf("error setting attribute `auth_key` after creating EVS volume transfer: %s", err)
}

Reply:
done